### PR TITLE
Add short test plans for OpenCL, Vulkan, and Level Zero (New)

### PR DESCRIPTION
**Pull request by mckees**
## Description

The current full test plans are useful for a deep dive on conformance, but they run far too long for a daily run or as part of the SRU process. These tests are trimmed down versions designed to run within 30-60 minutes on modern hardware.

## Resolved issues
N/A

## Documentation

N/A

## Tests

Run any of the launchers in providers/base/launcher/graphics/ to test.

### CODE REVIEW

File: providers/base/units/graphics/test-plan.yaml [hunk 1]
```diff
@@ -693,3 +693,124 @@ include:
...
+  - setup/install_vulkan_cts
+  - graphics/vk_gpu_avail
+  - graphics/vk_core_api
+  - graphics/vk_core_compute
+  - graphics/vk_core_draw
+  - graphics/vk_core_glsl
+  - graphics/vk_core_memory
+  - graphics/vk_core_renderpass
+  - graphics/vk_core_synchronization
+bootstrap_include:
```
[the remark this answers is not in the document]
Comment (mckees) [reply providers/base/units/graphics/test-plan.yaml:749]: Done

File: providers/base/units/graphics/test-plan.yaml [hunk 2]
```diff
@@ -693,3 +693,124 @@ include:
...
+  - graphics/cl_core_relationals
+  - graphics/cl_core_atomics
+  - graphics/cl_core_svm
+  - graphics/cl_sru_core_thread_dimensions
+  - graphics/cl_sru_core_integer_ops
+bootstrap_include:
+  - graphics_card
+  - executable
+---
+id: opencl-host-short-test-plan
```
Comment (tomli380576) [inline providers/base/units/graphics/test-plan.yaml:716]: optional: Can we rename this to `opencl-classic-short-test-plan` to stay consistent with the vulkan one?

**Review by tomli380576 — APPROVED** [review]
Overall LGTM, I was able to get all the test cases correctly generated on a lunar lake machine. Did you want to include the launchers in a separate PR? I saw you mention launchers in the PR description but it doesn't seem to be in checkbox's main branch.

**Comment by tomli380576** [comment]
Successful run (vulkan) https://certification.canonical.com/hardware/202408-35309/submission/508605/

**Comment by mckees** [comment]
> Overall LGTM, I was able to get all the test cases correctly generated on a lunar lake machine. Did you want to include the launchers in a separate PR? I saw you mention launchers in the PR description but it doesn't seem to be in checkbox's main branch.

I removed the launchers because there don't appear to be any in the base provider. I figured there must be a reason for that and will just add them on the fly in the CI

**Comment by baconYao** [comment]
Hi @mckees!

These plans are amazing! But I would like to know are these test plans also suitable for Ubuntu Core? 
Especially for some PE projects that need the specific GPU snap be installed in advanced.

**Comment by mckees** [comment]
> Hi @mckees!
> 
> These plans are amazing! But I would like to know are these test plans also suitable for Ubuntu Core? Especially for some PE projects that need the specific GPU snap be installed in advanced.

Hello! We are actually starting the work to expand to other architectures right now, so that will enable the snaps to run beyond Intel devices (and the CI to make sure that arch-specific regressions are caught early). @ahresse will handle this while I start merging our media tests. 

In terms of shorter test plans with a specific focus that is interesting to a particular team, my current plan is to leave that up to the teams in question to curate and submit for their own testing purposes. These short plans were mostly just about pulling stress tests, long-running tests, and (only in the case of Level Zero) tests which caused remote devices to drop and never reestablish their connection.

**Review by Hook25 — APPROVED** [review]
+1, Thanks

**Comment by Hook25** [comment]
Cancelling metabox and snap validation as they are expensive and the branch is just out of date, nothing to validate here.
